molo/surveys/rules.py

```diff
@@ -391,5 +392,35 @@ def description(self):
                 'particular combination of rules'),
         }
 
+    def clean(self):
+        super(CombinationRule, self).clean()
+        if isinstance(self.body.stream_data[0], dict):
+            newData = [block['type'] for block in self.body.stream_data]
+        elif isinstance(self.body.stream_data[0], tuple):
+            newData = [block[0] for block in self.body.stream_data]
+
+        if len(newData) % 3 != 0:
+            raise StreamBlockValidationError(non_block_errors=[_(
+                'Rule Combination must follow the <Rule/NestedLogic>'
+                '<Operator> <Rule/NestedLogic> pattern.')])
+
+        iterations = len(newData) / 3
+        print iterations
+        for i in range(iterations):
+            first_rule_index = (i * 3)
+            operator_index = (i * 3) + 1
+            second_rule_index = (i * 3) + 2
+
+            if((newData[first_rule_index] == 'Rule' or
+                newData[first_rule_index] == 'NestedLogic') and
+               (newData[operator_index] == 'Operator') and
+               (newData[second_rule_index] == 'Rule' or
+                    newData[second_rule_index] == 'NestedLogic')):
+                pass
```

Comment:
This is an interesting usage. I think the most logical flow would be a `if not <expr>: raise StreamBlockValidationError` but I'm not sure if it will improve readability. 🤔